### PR TITLE
Move classes to separate files

### LINKED ISSUE #468: Draft a proposal for a coherent code structuring approach within the project
Should we use just one app? What should be in one app, what in another?
Should we create one app and what sub-structure should we create inside?
Should we use e.g. only class based views, how should we group our classes?

Take a look at sucessful django projects and how they handle this? Keep in mind that a best-practice seemingly does not exist. I'd rather have a proposal that works for us and where we know where to expect a file. Take a look at what we implemendted so far and verify that the proposal defines a clear location for all code that we wrote

### PR DESCRIPTION
**Pull request by feeds**
Partly closes #468 

In this case I reordered the project by

- creating packages within each app for models, forms, views
- in each package every class gets its own file
- all these files are imported into the model automatically, just like in sentry via 'import_all' in the init
- that way one can still access all models via `from .models import Hospital` and likewise

I *did not and do not intend to* 
- reorder all views into class based views => seperate PR for that in #511
- set up a form folder in ineedstudent because of cyclic dependencies :angel: => seperate issue for the in #510 

I recommend checking out locally, because the file changed thing is super useless sin this case :grin:

### CODE REVIEW

**Comment by feeds** [comment]
The newsletter model contains a bug that is solved in #510

File: backend/match4healthcare/utils/imports.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import pkgutil
+
+import six
```
Comment (bjrne) [inline backend/match4healthcare/utils/imports.py:3]: Do we really need python2 compatibility? I would lobby against it and use python3 syntax.

File: backend/match4healthcare/utils/imports.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+
+def import_submodules(context, root_module, path):
+    """
+    Import all submodules and register them in the ``context`` namespace.
+
+    >>> import_submodules(locals(), __name__, __path__)
+    """
+    for loader, module_name, is_pkg in pkgutil.walk_packages(path, root_module + "."):
+        # this causes a Runtime error with model conflicts
```
Comment (bjrne) [inline backend/match4healthcare/utils/imports.py:16]: Is the occurence of this runtime error tested or just copied from the source? Maybe it is not an issue with python3 or stuff.

File: backend/apps/ineedstudent/views/__init__.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from match4healthcare.utils.imports import import_submodules
+
+import_submodules(globals(), __name__, __path__)
```
Comment (bjrne) [inline backend/apps/ineedstudent/views/__init__.py:3]: Let's test this with more people of our team whether this works with IDE linting or whether we need special configuration for this to work, which should then maybe be included in the repository.

**Review by bjrne — COMMENTED** [review]
Apart from the small comments, I really like it!

File: backend/apps/ineedstudent/views/__init__.py (hunk 3 above)
Comment (feeds) [reply backend/apps/ineedstudent/views/__init__.py:3]: Yes, maybe we can create a new issue for tihs? I think the understandability of the code with this refactoring improves so much that the code linting is not the bigger issue here.
At least I think it works with black, it kept correcting my commits for import style...

File: backend/apps/accounts/tables.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 from django.utils.translation import gettext_lazy as _
 import django_tables2 as tables
 
-from .models import Newsletter, NewsletterState
+from .modelss import Newsletter, NewsletterState
```
Comment (Baschdl) [inline backend/apps/accounts/tables.py]: 🙈

**Review by Baschdl — CHANGES_REQUESTED** [review]

File: backend/apps/ineedstudent/views/__init__.py (hunk 3 above)
Comment (kevihiiin) [reply backend/apps/ineedstudent/views/__init__.py:3]: My IDE also reports back that the modules (models) cannot be found. It also breaks the `manage.py createfakeusers` command for me.

Is there a certain order in which the submodules are loaded? If it is alphabetically the order might change when we add new one

- - - 
Edit: The error with manage.py was unrelated to the__init__.py file

Comment (kevihiiin) [reply backend/apps/ineedstudent/views/__init__.py:3]: Another (manual and not so elegant) alternative would be to explicitly list all the packages like so:
`from .hospital import *` 
or even better only import modules / variables that are actually used

This way the IDEs should be able to find the modules as it is not dynamically loaded and the order of imports is fixed (if that is important)

Suggested by the Django documentation:
https://docs.djangoproject.com/en/3.0/topics/db/models/#organizing-models-in-a-package

File: backend/apps/accounts/tables.py (hunk 4 above)
Comment (feeds) [reply backend/apps/accounts/tables.py]: das ist das letzte was geänder wird bevor es fertig ist :heart:

File: backend/apps/accounts/models/__init__.py [hunk 5]
```diff
@@ -1,3 +1,15 @@
 from match4healthcare.utils.imports import import_submodules
 
-import_submodules(globals(), __name__, __path__)
+STATIC_HACK = False
+if STATIC_HACK:
```
Comment (bjrne) [inline backend/apps/accounts/models/__init__.py:4]: LOL :rofl:

**Comment by maltezacharias** [comment]
@feeds How should we handle this atm? Should we
- close the PR?
- review & merge?
- do something else?

What state is this in?

**Comment by feeds** [comment]
it seemed like a too risky change to apply this to the repo. Wer just users it in m4e.

so i think this can be closed

**Comment by maltezacharias** [comment]
Major changes are for now not planned to be implemented in m4h. 
A more complete redesign using these concepts has been applied to m4e

**Pull request closed without merging**